src/callback_requests.rs

```diff
@@ -357,3 +365,56 @@ impl From<kubewarden_policy_sdk::host_capabilities::kubernetes::GetResourceReque
         }
     }
 }
+
+impl From<SubjectAccessReview> for CallbackRequestType {
+    fn from(req: SubjectAccessReview) -> Self {
+        CallbackRequestType::KubernetesCanI {
+            subject_access_review: Box::new(SubjectAccessReviewWrapper(req)),
+        }
+    }
+}
+
+/// Wrapper type to allow users to pass a `SubjectAccessReview` to the
+/// `KubernetesCanI` request. The original types does not implement the Eq trait.
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct SubjectAccessReviewWrapper(pub SubjectAccessReview);
+
+impl Eq for SubjectAccessReviewWrapper {}
+
+/// Implementing the AsRef trait for SubjectAccessReviewWrapper to allow
+/// easy access to the inner SubjectAccessReview type.
+impl AsRef<SubjectAccessReview> for SubjectAccessReviewWrapper {
+    fn as_ref(&self) -> &SubjectAccessReview {
+        &self.0
+    }
+}
+
+/// Implementing the Display trait for SubjectAccessReviewWrapper to simplify the
+/// cache key generation. This implementation uses a hasher to generate a unique
+/// identifier for the SubjectAccessReviewWrapper instance.
+impl Display for SubjectAccessReviewWrapper {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut hasher = DefaultHasher::new();
+        self.hash(&mut hasher);
```

Comment:
how does the hash work? The SubjectAccessReview.spec has the groups, which is an array. These groups may be in a different order. Does this mean it's a different cache key?

Not a bug, just less efficient when finding cache hits.